### PR TITLE
Add Data sources viewing support in the dashboard

### PR DESCRIPTION
**Pull request by SanojPunchihewa**
## Purpose
Fixes https://github.com/wso2/product-mi-tooling/issues/233
Fixes https://github.com/wso2/product-mi-tooling/issues/234
Fixes https://github.com/wso2/micro-integrator/issues/1812

## Goals
This feature will improve the MI dashboard to view Data sources.

### CODE REVIEW

File: monitoring-dashboard/components/org.wso2.micro.integrator.dashboard.web/web-app/src/commons/sideDrawers/DataSourcesSideDrawer.js [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+        <TableRow>
+            <TableCell>Username</TableCell>
+            <TableCell>{nodeData.userName}</TableCell>
+        </TableRow>
+    </Table>
+}
+
+function ParametersSection(props) {
+    const parameters = props.parameters;
+    return <Table size="small" style={{width: '100%'}}>
```
Comment (tharakamd) [inline monitoring-dashboard/components/org.wso2.micro.integrator.dashboard.web/web-app/src/commons/sideDrawers/DataSourcesSideDrawer.js:92]: `{{width: '100%'}}` shall we define this style also in the `useStyles()` function ?

File: monitoring-dashboard/components/org.wso2.micro.integrator.dashboard.web/web-app/src/commons/sideDrawers/DataSourcesSideDrawer.js [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+    const classes = useStyles();
+
+    return (
+        <div className={classes.root}>
+            <Grid container spacing={3}>
+                <Grid item xs={12}>
+                    <HeadingSection name={artifactName} nodeId={nodeId} />
+                    <SourceViewSection
+                        artifactType="data-sources" artifactName={artifactName} nodeId={nodeId}
+                        designContent={<>
```
Comment (tharakamd) [inline monitoring-dashboard/components/org.wso2.micro.integrator.dashboard.web/web-app/src/commons/sideDrawers/DataSourcesSideDrawer.js:44]: Can we move this section to a new function to improve the readability of the code?

File: monitoring-dashboard/components/org.wso2.micro.integrator.dashboard.web/web-app/src/pages/Datasources.js [hunk 3]
```diff
@@ -20,38 +20,30 @@
...
-                tableOrderBy: 'service'
-            },
-            datasourceList: [{
-                name: "Calculator EP",
-                nodes: [
-                    { nodeId: "node_01",
-                        type: "Type",
-                        data_source_status: "Inactive"
+export default function DataSources() {
+    const [pageInfo] = React.useState({
```
Comment (tharakamd) [inline monitoring-dashboard/components/org.wso2.micro.integrator.dashboard.web/web-app/src/pages/Datasources.js:27]: Shall we refactor these `React.useState()` to `useState()` (by fixing imports) to maintain consistency. I mean all the other hooks are called like `useSelector()`, so it's better to use the same format for the `useState()` as well.

Comment (tharakamd) [reply monitoring-dashboard/components/org.wso2.micro.integrator.dashboard.web/web-app/src/pages/Datasources.js:27]: `useEffect()` in line 42 need the same modification.

File: monitoring-dashboard/components/org.wso2.micro.integrator.dashboard.web/web-app/src/commons/sideDrawers/DataSourcesSideDrawer.js [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+            <TableCell>{nodeData.userName}</TableCell>
+        </TableRow>
+    </Table>
+}
+
+function ParametersSection(props) {
+    const parameters = props.parameters;
+    return <Table size="small" style={{width: '100%'}}>
+        {Object.entries(parameters).map(([key, value]) => 
+            <TableRow>
```
Comment (tharakamd) [inline monitoring-dashboard/components/org.wso2.micro.integrator.dashboard.web/web-app/src/commons/sideDrawers/DataSourcesSideDrawer.js:94]: Shall we try to define a key for this loop [1]? We might be getting a small and less frequently updating list now, but if we have defined a key, if this list grows in the future, there won't be any performance degradation. 

[1] - https://reactjs.org/docs/lists-and-keys.html#keys

Comment (SanojPunchihewa) [reply monitoring-dashboard/components/org.wso2.micro.integrator.dashboard.web/web-app/src/commons/sideDrawers/DataSourcesSideDrawer.js]: Fixed with 795cc4d

File: monitoring-dashboard/components/org.wso2.micro.integrator.dashboard.web/web-app/src/pages/Datasources.js (hunk 3 above)
Comment (SanojPunchihewa) [reply monitoring-dashboard/components/org.wso2.micro.integrator.dashboard.web/web-app/src/pages/Datasources.js]: Fixed with 795cc4d

File: monitoring-dashboard/components/org.wso2.micro.integrator.dashboard.web/web-app/src/commons/sideDrawers/DataSourcesSideDrawer.js (hunk 2 above)
Comment (SanojPunchihewa) [reply monitoring-dashboard/components/org.wso2.micro.integrator.dashboard.web/web-app/src/commons/sideDrawers/DataSourcesSideDrawer.js]: Fixed with 795cc4d

File: monitoring-dashboard/components/org.wso2.micro.integrator.dashboard.web/web-app/src/commons/sideDrawers/DataSourcesSideDrawer.js (hunk 1 above)
Comment (SanojPunchihewa) [reply monitoring-dashboard/components/org.wso2.micro.integrator.dashboard.web/web-app/src/commons/sideDrawers/DataSourcesSideDrawer.js]: Fixed with 795cc4d

**Review by tharakamd — APPROVED** [review]

**Pull request merged**